shuup/front/checkout/confirm.py

```diff
@@ -123,9 +149,16 @@ def form_valid(self, form):
     def create_order(self):
         basket = self.basket
         assert basket.shop == self.request.shop
-        basket.orderer = self.request.person
-        basket.customer = self.request.customer
-        basket.creator = self.request.user
+        gdpr_settings = GDPRSettings.get_for_shop(basket.shop)
+        if type(self.request.person) == AnonymousContact and has_installed("shuup.gdpr") and gdpr_settings.enabled:
```

Comment:
Maybe something like this would look nicer.

1. you do not need type check
2. you should deal gdpr settings if gdpr is installed
```
contact, user = None
if not self.request.person and has_installed("shuup.gdpr"):
    from shuup.gdpr.models import GDPRSettings
    gdpr_settings = GDPRSettings.get_for_shop(basket.shop)
    if gdpr_settings.enabled:
          contact, user = self._get_guest_order_data_with_gdpr()  

basket.orderer = (self.request.person or contact)
basket.customer = (self.request.customer or contact)
basket.creator = (self.request.user or user)
```